lib/Cake/Network/CakeSocket.php

```diff
@@ -170,6 +170,25 @@ public function connect() {
 		if ($this->connected) {
 			stream_set_timeout($this->connection, $this->config['timeout']);
 		}
+
+		if (!empty($this->config['request']) && $this->config['request']['uri']['scheme'] === 'https' && !empty($this->config['proxy'])) {
+			$req = array();
+			$req[] = 'CONNECT '. $this->config['request']['uri']['host'] . ':' . $this->config['request']['uri']['port'] . ' HTTP/1.1';
+			$req[] = 'Host: ' . $this->config['host'];
+			$req[] = 'User-Agent: php proxy';
+
+			fwrite($this->connection, implode("\r\n", $req)."\r\n\r\n");
+
+			while(true) {
+				$s = rtrim(fgets($this->connection, 4096));
+				if (preg_match('/^$/', $s)) {
+					break;
+				}
```

Comment:
What happens if this case never happens. You're in an infinite loop. Instead should you be checking `while (feof($this->connection)) {`?